extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigMappingCreator.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.arc.runtime;
+
+import java.util.Map;
+
+import javax.enterprise.context.spi.CreationalContext;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.arc.BeanCreator;
+import io.smallrye.config.SmallRyeConfig;
+
+public class ConfigMappingCreator implements BeanCreator<Object> {
```

Comment:
Shouldn't this generic parameter be more specific?

Reply:
No. This is the equivalent to the `@ConfigProperties` feature in SmallRye. It uses a plain interface class to group properties.